zipkin-collector/kafka10/src/main/java/zipkin/collector/kafka10/KafkaCollector.java

```diff
@@ -0,0 +1,223 @@
+import static zipkin.internal.Util.checkNotNull;
+
+/**
+ * This collector polls a Kafka topic for messages that contain TBinaryProtocol big-endian encoded
+ * lists of spans. These spans are pushed to a {@link AsyncSpanConsumer#accept span consumer}.
+ *
+ * <p>This collector uses a Kafka 0.10+ consumer.
+ */
+public final class KafkaCollector implements CollectorComponent {
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaCollector.class);
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /** Configuration including defaults needed to consume spans from a Kafka topic. */
+  public static final class Builder implements CollectorComponent.Builder {
+    final Properties properties = new Properties();
+    Collector.Builder delegate = Collector.builder(KafkaCollector.class);
+    CollectorMetrics metrics = CollectorMetrics.NOOP_METRICS;
+    String topic = "zipkin";
+    int streams = 1;
+
+    @Override public Builder storage(StorageComponent storage) {
+      delegate.storage(storage);
+      return this;
+    }
+
+    @Override public Builder sampler(CollectorSampler sampler) {
+      delegate.sampler(sampler);
+      return this;
+    }
+
+    @Override public Builder metrics(CollectorMetrics metrics) {
+      this.metrics = checkNotNull(metrics, "metrics").forTransport("kafka");
+      delegate.metrics(this.metrics);
+      return this;
+    }
+
+    /** Topic zipkin spans will be consumed from. Defaults to "zipkin" */
+    public Builder topic(String topic) {
+      this.topic = checkNotNull(topic, "topic");
+      return this;
+    }
+
+    /** The bootstrapServers connect string, ex. 127.0.0.1:9092. No default. */
+    public Builder bootstrapServers(String bootstrapServers) {
+      properties.put(BOOTSTRAP_SERVERS_CONFIG, checkNotNull(bootstrapServers, "bootstrapServers"));
+      return this;
+    }
+
+    /** The consumer group this process is consuming on behalf of. Defaults to "zipkin" */
+    public Builder groupId(String groupId) {
+      properties.put(GROUP_ID_CONFIG, checkNotNull(groupId, "groupId"));
+      return this;
+    }
+
+    /** Count of threads consuming the topic. Defaults to 1 */
+    public Builder streams(int streams) {
+      this.streams = streams;
+      return this;
+    }
+
+    /**
+     * By default, a consumer will be built from properties derived from builder defaults,
+     * as well as "auto.offset.reset" -> "earliest". Any properties set here will override the
+     * consumer config.
+     *
+     * <p>For example: Only consume spans since you connected by setting the below.
+     * <pre>{@code
+     * Map<String, String> overrides = new LinkedHashMap<>();
+     * overrides.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
+     * builder.overrides(overrides);
+     * }</pre>
+     *
+     * @see org.apache.kafka.clients.consumer.ConsumerConfig
+     */
+    public final Builder overrides(Map<String, String> overrides) {
```

Comment:
I ran out of steam before getting to this. If anybody feels strongly about using autovalue here, let me know and I will circle back around to it.